core/user-agent.c

```diff
@@ -232,12 +232,8 @@ ua_conn_add_header(struct ua_conn *conn,
             && 0 == strncasecmp(node->data, field, fieldlen))
         {
             if (strlen(node->data) < buflen) {
-                /* FIXME: For some reason, cygwin builds will abort on this
-                 * free() */
-#ifndef __CYGWIN__
-                free(node->data);
-#endif
-                node->data = strdup(buf);
+                curl_free(node->data);
+                curl_slist_append(NULL, buf);
```

Comment:
[`curl_slist_append`](https://curl.se/libcurl/c/curl_slist_append.html) with a NULL as slist is used to create a new slist, but that would be only if we were getting its return value.

We would either need to watch its return value or replace it

Reply:
Fixed!